src/cpp/rtps/builtin/BuiltinProtocols.cpp

```diff
@@ -308,7 +308,10 @@ void BuiltinProtocols::stopRTPSParticipantAnnouncement()
 
     if (mp_PDP)
     {
-        mp_PDP->stopParticipantAnnouncement();
+        if (mp_PDP->is_enabled())
```

Comment:
I would put this check inside `PDP::stopParticipantAnnouncement`. And then, I don't think we will need `PDP::is_enabled`.